lib/structures/ComponentInteraction.js

```diff
@@ -0,0 +1,275 @@
+    * @arg {Buffer} content.file.file A buffer containing file data
+    * @arg {String} content.file.name What to name the file
+    * @arg {Boolean} [content.tts] Set the message TTS flag
+    * @returns {Promise<Message?>}
+    */
+    async createFollowup(content) {
+        if(content !== undefined) {
+            if(typeof content !== "object" || content === null) {
+                content = {
+                    content: "" + content
+                };
+            } else if(content.content !== undefined && typeof content.content !== "string") {
+                content.content = "" + content.content;
+            }
+        }
+        return this._client.executeWebhook.call(this._client, this.applicationID, this.token, Object.assign({wait: true}, content));
+    }
+
+    /**
+    * Acknowledges the interaction with a message
+    * Note: You can **not** use more than 1 initial interaction response per interaction, use createFollowup if you have already responded with a different interaction response.
+    * @arg {String | Object} content A string or object. If an object is passed:
+    * @arg {Object} [content.allowedMentions] A list of mentions to allow (overrides default)
+    * @arg {Boolean} [content.allowedMentions.everyone] Whether or not to allow @everyone/@here.
+    * @arg {Boolean | Array<String>} [content.allowedMentions.roles] Whether or not to allow all role mentions, or an array of specific role mentions to allow.
+    * @arg {Boolean | Array<String>} [content.allowedMentions.users] Whether or not to allow all user mentions, or an array of specific user mentions to allow.
+    * @arg {String} content.content A content string
+    * @arg {Object} [content.embeds] An array of up to 10 embed objects. See [the official Discord API documentation entry](https://discord.com/developers/docs/resources/channel#embed-object) for object structure
+    * @arg {Number} [content.flags] 64 for Ephemeral
+    * @arg {Boolean} [content.tts] Set the message TTS flag
+    * @returns {Promise}
+    */
+    async createMessage(content) {
+        if(content !== undefined) {
+            if(typeof content !== "object" || content === null) {
+                content = {
+                    content: "" + content
+                };
+            } else if(content.content !== undefined && typeof content.content !== "string") {
+                content.content = "" + content.content;
+            } else if(content.content === undefined && !content.embeds) {
+                return Promise.reject(new Error("No content or embeds"));
+            }
+            if(content.content !== undefined || content.embeds || content.allowedMentions) {
+                content.allowed_mentions = this._client._formatAllowedMentions(content.allowedMentions);
+            }
+        }
+        return this._client.createInteractionResponse.call(this._client, this.id, this.token, {
+            type: InteractionResponseTypes.CHANNEL_MESSAGE_WITH_SOURCE,
```

Comment:
This is the wrong interaction response type for components
```suggestion
            type: InteractionResponseTypes.UPDATE_MESSAGE,
```

Reply:
I forgot to add the other methods as well, but it looks like all of the ones in ComponentInteraction.js are using slash command response types instead of component response types

Reply:
Defer is the same as the slash commands one, deferUpdate is not, I just forgot to change acknowledge back to what it was

Reply:
Also I think the method was defer_update_message

Reply:
Wait no that's correct I think, component interactions use the same responses, but have a few extra, aka editParent and deferUpdate

Reply:
Just checked, it is already correct